File: backend/src/main/java/com/example/bsuir/character/service/CharacterService.java
package com.example.bsuir.character.service;

import com.example.bsuir.character.dto.request.CreateCharacterRequest;
import com.example.bsuir.character.dto.request.UpdateCharacterRequest;
import com.example.bsuir.character.model.Character;
import com.example.bsuir.shared.service.Service;

import java.util.List;

public interface CharacterService extends Service<Character, Long> {

    List<Character> getAllByShowId(Long showId);

    Character getByImageId(Long imageId);

    Character create(CreateCharacterRequest request);

    Character updateById(Long id, UpdateCharacterRequest request);

    void deleteById(Long id);
}

File: frontend/src/character/service/CharacterService.js
import axios from 'axios';

export default class CharacterService {

    getCharacterRestApiUrl(showId) {
        return `http://localhost:8080/show/${showId}/character`;
    }

    getAllByShowId(showId) {
        const url = this.getCharacterRestApiUrl(showId);
        return axios.get(url);
    }

    getById(showId, id) {
        const url = `${this.getCharacterRestApiUrl(showId)}/${id}`;
        return axios.get(url);
    }

    create(showId, request) {
        const url = this.getCharacterRestApiUrl(showId);
        return axios.post(url, request);
    }

    updateById(showId, id, request) {
        const url = `${this.getCharacterRestApiUrl(showId)}/${id}`;
        return axios.put(url, request);
    }

    deleteById(showId, id) {
        const url = `${this.getCharacterRestApiUrl(showId)}/${id}`;
        return axios.delete(url);
    }
}
